src/modules/groupAccess.ts

```diff
@@ -1,11 +1,38 @@
+import { GroupAdmin } from 'prisma/prisma-client';
+
+import { prisma } from '../utils/prisma';
 import { AccessCheckResult, allowed, notAllowed } from '../utils/access';
 import { SessionUser } from '../utils/auth';
 
 import { tr } from './modules.i18n';
 
+const checkGroupAdmin = async (groupId: string, userId: string) => {
+    const admins = await prisma.$queryRaw<GroupAdmin[]>`
+    SELECT count(*) FROM public."GroupAdmin"
+    WHERE "userId" = ${userId}
+    AND "groupId" in (
+        WITH RECURSIVE rectree AS (
+            SELECT *
+            FROM public."Group"
+            WHERE id = ${groupId}
+        UNION ALL
+            SELECT g.*
+            FROM public."Group" g
+            JOIN rectree
+                ON g.id = rectree."parentId"
+        ) SELECT id FROM rectree
+    )
+    `;
+
+    return admins.length > 0;
```

Comment:
if you select count, you shouldn't check length, it will return number. Also, type casting is wrong